sparse_strips/vello_hybrid/src/render_webgl.rs

```diff
@@ -0,0 +1,1034 @@
+                0,
+                WebGl2RenderingContext::RGBA8 as i32,
+                render_size.width as i32,
+                render_size.height as i32,
+                0,
+                WebGl2RenderingContext::RGBA,
+                WebGl2RenderingContext::UNSIGNED_BYTE,
+                None,
+            )
+            .expect("Failed to resize view texture");
+
+            self.render_size = render_size.clone();
+        }
+
+        // Process alpha data for texture
+        if !alphas.is_empty() {
+            let texture_width = self.resources.alpha_texture_width;
+            let texture_height = self.resources.alpha_texture_height;
+
+            debug_assert!(
+                alphas.len() <= (texture_width * texture_height * 16) as usize,
+                "Alpha texture dimensions are too small to fit the alpha data"
+            );
+
+            // After this copy to `self.alpha_data`, there may be stale trailing alpha values. These
+            // are not sampled, so can be left as-is.
+            self.alpha_data[0..alphas.len()].copy_from_slice(alphas);
+            gl.active_texture(WebGl2RenderingContext::TEXTURE0);
+            gl.bind_texture(
+                WebGl2RenderingContext::TEXTURE_2D,
+                Some(&self.resources.alphas_texture),
+            );
+
+            // Pack alpha values into RGBA uint32 texture
+            let alpha_data_as_u32 = bytemuck::cast_slice::<u8, u32>(&self.alpha_data);
+            let packed_array = js_sys::Uint32Array::from(alpha_data_as_u32);
+
+            gl.tex_image_2d_with_i32_and_i32_and_i32_and_format_and_type_and_opt_array_buffer_view(
+                WebGl2RenderingContext::TEXTURE_2D,
+                0,
+                WebGl2RenderingContext::RGBA32UI as i32,
+                texture_width as i32,
+                texture_height as i32,
+                0,
+                WebGl2RenderingContext::RGBA_INTEGER,
+                WebGl2RenderingContext::UNSIGNED_INT,
+                Some(&packed_array),
+            )
+            .expect("Failed to write alpha texture");
+        }
+
+        // Clear the view framebuffer.
+        {
+            gl.bind_framebuffer(
+                WebGl2RenderingContext::FRAMEBUFFER,
+                Some(&self.resources.view_framebuffer),
+            );
+            gl.clear_color(0.0, 0.0, 0.0, 0.0);
+            gl.clear(WebGl2RenderingContext::COLOR_BUFFER_BIT);
+        }
+    }
+
+    /// Upload strip data to GPU.
+    fn upload_strips(&mut self, gl: &WebGl2RenderingContext, strips: &[GpuStrip]) {
+        if strips.is_empty() {
+            return;
+        }
+
+        gl.bind_buffer(
+            WebGl2RenderingContext::ARRAY_BUFFER,
+            Some(&self.resources.strips_buffer),
+        );
+
+        let strips_data = bytemuck::cast_slice(strips);
+        let strips_array = js_sys::Uint8Array::from(strips_data);
+
+        gl.buffer_data_with_array_buffer_view(
```

Comment:
Should this be:

```suggestion
        gl.buffer_data_with_js_u8_array(
```

Do you know whether it's more advantageous to do the below (and skip creating the `Uint8Array`)?

```suggestion
        gl.buffer_data_with_u8_array(
```

(Same goes for other buffer uploads)

Reply:
I wasn't able to figure out if there's any cost, but there probably is some cost in creating the wrapped JavaScript object.  I simplified all usages of `buffer_data_with_array_buffer_view` to instead use `buffer_data_with_u8_array`.

Great find!